src/adapters/sqlite/integrationTest.js

```diff
@@ -1,18 +1,47 @@
+import expect from 'expect'
+import { Platform } from 'react-native'
 import SQLiteAdapter from './index'
 import { testSchema } from '../__tests__/helpers'
 import commonTests from '../__tests__/commonTests'
-
-const newAdapter = () => new SQLiteAdapter({ schema: testSchema })
+import { invariant } from '../../utils/common'
 
 const SQLiteAdapterTest = spec => {
-  spec.describe('SQLiteAdapter', () => {
+  const runTests = isSynchronous => {
     commonTests().forEach(testCase => {
       const [name, test] = testCase
       spec.it(name, async () => {
-        const adapter = newAdapter()
+        const adapter = new SQLiteAdapter({ schema: testSchema, synchronous: isSynchronous })
+
+        if (isSynchronous) {
+          if (Platform.OS === 'ios') {
+            invariant(adapter._synchronous === true, 'this should be synchronous')
+          } else {
+            invariant(
+              adapter._synchronous === false,
+              'this should be asynchronous - android does not support synchronous adaoter',
```

Comment:
```suggestion
              'this should be asynchronous - android does not support synchronous adapter',
```

Reply:
Also, we should avoid creating such discrepancy between Android and iOS. We all know how frustrating is when you are trying to implement something but for some reason it works only on one platform

Reply:
True, OTOH: a) we agreed on Slack that it's not worth pursuing this internally on Android with current performance levels, b) it's an option you have to know about to use, c) there's a graceful fallback